Automattic/wp-calypso
# Use the v1.3 site search endpoint

#### Changes proposed in this Pull Request

- [x] add fetch layer for the v1.3 POST site search endpoint
- [x] send all ES query parameters needed for marketplace products
- [x] replace the api call in the site search hook
- [x] map data to plugin browser pages schema
- [x] ~filter by author~ It will be done as part of https://github.com/Automattic/wp-calypso/issues/64801
- [x] apply language parameter
- [x] ~use wpcom client~ It will be done as part of https://github.com/Automattic/wp-calypso/issues/64802

#### Testing instructions



* Enable the feature flag `marketplace-jetpack-plugin-search`
* Go to `/plugins/<your-site>`, add a search term and press Enter
* Make sure you can see results



Fixes #63166
